custom-targets/util/clouddeploy/build_and_register.sh

```diff
@@ -0,0 +1,121 @@
+# Copyright 2023 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https:#www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#!/bin/bash
+
+set -e
+
+if [[ ! -v _CT_SRCDIR || ! -v _CT_IMAGE_NAME || ! -v _CT_TYPE_NAME || ! -v _CT_CUSTOM_ACTION_NAME || ! -v _CT_GCS_DIRECTORY || ! -v _CT_SKAFFOLD_CONFIG_NAME ]]; then
+  echo "This script is not meant to be used on its own. Please launch it from one of the custom target directories."
+  exit 1
+fi
+
+usage() {
+  echo "usage: build_and_register.sh -p <project_id> -r <region>"
+}
+
+boldout() {
+  echo $(tput bold)$(tput setaf 1)"$@"$(tput sgr0)
+}
+
+while getopts "p:r:" arg; do
+  case "${arg}" in
+    p)
+      PROJECT="${OPTARG}"
+      ;;
+    r)
+      REGION="${OPTARG}"
+      ;;
+    *)
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+if [[ ! -v PROJECT || ! -v REGION ]]; then
+  usage
+  exit 1
+fi
+
+AR_REPO=$REGION-docker.pkg.dev/$PROJECT/cd-custom-targets
+if ! gcloud -q artifacts repositories describe --location "$REGION" --project "$PROJECT" cd-custom-targets > /dev/null 2>&1; then
+  boldout "Creating Artifact Registry repository: ${AR_REPO}"
+  gcloud -q artifacts repositories create --location "$REGION" --project "$PROJECT" --repository-format docker cd-custom-targets
+fi
+
+boldout "Granting the default compute service account access to ${AR_REPO}"
+gcloud -q artifacts repositories add-iam-policy-binding \
+    --project "${PROJECT}" --location "${REGION}" cd-custom-targets \
+    --member=serviceAccount:$(gcloud -q projects describe $PROJECT --format="value(projectNumber)")-compute@developer.gserviceaccount.com \
+    --role="roles/artifactregistry.reader" > /dev/null
+
+BUCKET_NAME="${PROJECT}-${REGION}-custom-targets"
+if ! gsutil ls "gs://${BUCKET_NAME}" > /dev/null 2>&1; then
+  boldout "Creating a storage bucket to hold the custom target configuration"
+  gcloud -q storage buckets create --project "${PROJECT}" --location "${REGION}" "gs://${BUCKET_NAME}"
+fi
+
+boldout "Building the Custom Target image in Cloud Build."
+boldout "This will take approximately 10 minutes"
+
+# get the commit hash to pass to the build
+COMMIT_SHA=$(git rev-parse --verify HEAD)
+
+CLOUDBUILD_YAML="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )/cloudbuild.yaml"
+# TODO(plumpy): just pass the URL to the cloudbuild.yaml file in Github, once it's public
```

Comment:
Remove this TODO. I think the intent before was to not have the clone the repo, but it's kind of necessary in order to run the script and any of the quickstarts since the configuration is in this repo.

Reply:
Done